### DISCUSSION TITLE
Does mfem support 2nd order triangular elements?

### DISCUSSION THREAD

**Discussion by AlemayehuB** [mfem/mfem#2440] [Q&A] [answered]
Hello, I'm new user and have been creating meshes of simple 2D shapes with netgen and gmsh that implement  2nd order triangular elements. However when I run glvis on these meshes I encounter a segmentation fault.

netgen> Mesh -> Meshing Options:
                   select "Second order elements"
                   select "Mesh granularity" = very fine
 netgen> Mesh -> Generate Mesh
 netgen> File -> Export mesh -> <name.mesh>

ng2-to-mfem.sh <name.mesh> <newname.mesh>

I've done the same thing without selecting "Second order elements" and glvis runs without any issues.

The following is a sample input that gave me issues:

`
areamesh2

8
   1            1        5       11
   1            5        2       12
   2            2        6       13
   2            6        3       14
   3            3        7       15
   3            7        4       16
   4            4        8       17
   4            8        1       18
10
        1        3        1        5        9
        1        3        2        9        5
        1        3        2        6       10
        1        3        2       10        9
        1        3        3       10        6
        1        3        3        7       10
        1        3        4       10        7
        1        3        4        8       10
        1        3        1        9        8
        1        3        8        9       10
29
  0.000000 -1.000000 
  1.000000  0.000000 
  0.000000  1.000000 
 -1.000000  0.000000 
  0.707107 -0.707107 
  0.707107  0.707107 
 -0.707107  0.707107 
 -0.707107 -0.707107 
  0.151753 -0.409423 
 -0.072105  0.222714 
  0.368095 -0.929788 
  0.929788 -0.368095 
  0.929788  0.368095 
  0.368095  0.929788 
 -0.368095  0.929788 
 -0.929788  0.368095 
 -0.929788 -0.368095 
 -0.368095 -0.929788 
  0.429430 -0.558265 
  0.075877 -0.704711 
  0.575877 -0.204711 
  0.317501  0.464910 
  0.463947  0.111357 
  0.039824 -0.093354 
 -0.036053  0.611357 
 -0.389606  0.464910 
 -0.536053  0.111357 
 -0.389606 -0.242196 
 -0.277677 -0.558265
 `

Thank You!

**Comment by tzanio** [comment, marked as answer]
Yes we do support second order (actually arbitrary) order triangular elements.

I don't believe we have tested NETGEN in a while though, and their format may have changed. Do you mind sharing the input file and which NETGEN version you used?

Tzanio

**Reply by AlemayehuB** [reply]
Hi I think the issue might have been on my end.

Thank You!
